custom_components/waste_collection_schedule/waste_collection_schedule/source/seattle_gov.py

```diff
@@ -21,40 +21,132 @@ def __init__(self, street_address):
+        r = requests.post(
+            f"https://myutilities.seattle.gov/rest/serviceorder/findAccount",
+            json=find_account_payload)
+
+
+        account_info = json.loads(r.text)
+        account_number = account_info["account"]["accountNumber"]
+
+
+
+        #step 3
+        token_payload = {"grant_type": "password", "username":"guest","password":"guest"}
+
+        r = requests.post(
+            f"https://myutilities.seattle.gov/rest/auth/token",
+            data=token_payload)
+       
+        token_info = json.loads(r.text)
+        token = token_info["access_token"]
+
+        headers = {
+            "Authorization": "Bearer {0}".format(token)
+        }
+
+        #step 4
+        swsummary_payload = {
+            "customerId": "guest",
+            "accountContext": {
+                "accountNumber": account_number,
+                "personId": None,
+                "companyCd": None,
+                "serviceAddress": None
+            }
+        }
+
+        r = requests.post(
+            f"https://myutilities.seattle.gov/rest/account/swsummary",
+            json=swsummary_payload,
+            headers=headers)
+
+
+    
+        summary_info = json.loads(r.text)
+        # the description property in each service in swServices it's either 'Garbage', 'Recycle', or 'Food/Yard Waste'
+        swServices = summary_info["accountSummaryType"]["swServices"][0]["services"]
+        personId = summary_info["accountContext"]["personId"]
+        companyCd = summary_info["accountContext"]["companyCd"]
+
+
+        #step 5
+        waste_calendar_payload = {
+            "customerId": "guest",
+            "accountContext": {
+                "accountNumber": account_number,
+                "personId": personId,
+                "companyCd": companyCd
+            },
+            "servicePoints": []
+        }
+
+        #fill out payload
+        for service in swServices:
+            waste_calendar_payload["servicePoints"].append(service["servicePointId"])
+
+
+        r = requests.post(
+            f"https://myutilities.seattle.gov/rest/solidwastecalendar",
+            json=waste_calendar_payload,
+            headers=headers)
+
+
+        calendar_info = {}
         entries = []
 
-        if next_pickup["Garbage"]:
-            entries.append(
-                Collection(date=next_pickup_date, t="Trash", icon="mdi:trash-can")
-            )
-        if next_pickup["FoodAndYardWaste"]:
-            entries.append(
-                Collection(
-                    date=next_pickup_date, t="Food and Yard Waste", icon="mdi:leaf"
+
+        calendar_info = json.loads(r.text)
+
+        for service in swServices:
+            name = service["description"]
+            servicePointId = service["servicePointId"]
+            next_date = datetime.datetime.strptime(calendar_info["calendar"][servicePointId+"_NP"][0], "%m/%d/%Y").date()
+            
+            if name == "Garbage":
```

Comment:
I can use the original name but I assume I still need the if/elif to specify the icon to use. Is that not accurate? Do you have a better example I can follow?

Reply:
sure. this would mean that this is a breaking change for anyone who is using the Seattle source and has customized it with the current names, correct?

Reply:
fixed in the latest commit.

Reply:
Yes, you're right. This may break existing waste type filters in sensor configurations.